esp/mlrs-wireless-bridge/mlrs-wireless-bridge-boards.h

```diff
@@ -108,8 +114,28 @@ GPIO15 = RTC_GPIO13
 // Module details
 //-------------------------------------------------------
 
+//-- Generic ESP8266, used in many ELRS Tx modules
+#if defined MODULE_GENERIC_ESP8266
+    #ifndef ARDUINO_ESP8266_GENERIC // ARDUINO_BOARD != ESP8266
+	      #error Select board Generic ESP8266 module
+    #endif
+
+    #undef USE_SERIAL_DBG1
+    #undef USE_SERIAL1_DBG
+    #undef USE_SERIAL2_DBG
+
+    #undef LED_IO
+    #define LED_IO 16
+    #define USE_LED
+
+    #undef GPIO0_IO
+    #define GPIO0_IO  0
```

Comment:
I suppose I should leave out GPIO0_ID here so it can be defined or not in the .ino file or create a second module without it.  Which do you prefer?

Reply:
I'm sorry but I had no time yet to look at it and without it's not clear what you exactly mean and what the issue at hand is.
If "second module" means yet another .h file, I guess I would less prefer that.

Reply:
No, module as in another esp8266 variant "board/module" section right here.

Reply:
ok, I guess I see your point. If I understand correctly, I would suggest this:
The generic case I think is
- not needed for now
- may work with the existing GENERIC
- and if not may either made to work with the existing GENERIC or one could add a esp8266 generic later
So, I guess my suggestion to simply just rename the current GENERIC_8266 to something like e.g. MODULE_ESP8266_INTERNAL_TX

Reply:
as a general comment I guess I would prefer to have, in e.g. the #if splices, the EP8266 to come as last, so rather somthing like #ifndef ESP8266.
I also don't quite comprehend yet why e.g. the inlcudes had to be moved up, if possible I would prefer them to stay where they are.
I also don't quite comprehend yet why the "#ifdef GPIO0_IO #define USE_AT_MODE #endif" piece did go away.

Reply:
The esp8266 is also configurable on many external ELRS Tx modules, so I think MODULE_ESP8266_ELRS_TX instead of MODULE_ESP8266_INTERNAL_TX

Reply:
When both the #ifdef and #else clauses are present, I really perfer ifdef over ifndef.  Let me know if you feel strongly about the order.

The reason I moved the #includes is because one of them is needed to define the IPAddress type.

The "#ifdef GPIO0_IO #define USE_AT_MODE #endif" piece is still there.

Reply:
MODULE_ESP8266_INTERNAL_TX is good

order, hm. would have liked to have esp32 dominance ... hmhm

> The reason I moved the #includes is because one of them is needed to define the IPAddress type.
which one woud it be which would be need newly now?

Reply:
I'm getting mixed signals from you here.  You gave thumbs up to MODULE_ESP8266_ELRS_TX and then said MODULE_ESP8266_INTERNAL_TX is good.  I'm confused enough on my own!

Reply:
> which one woud it be which would be need newly now?

ESP8266WiFi.h  It also just more common to include standard system headers before jumping into defining variables, etc.

Reply:
> order, hm. would have liked to have esp32 dominance ... hmhm

Try to think of it as esp32 getting the last word.  :)

Reply:
> I'm getting mixed signals from you here. You gave thumbs up to MODULE_ESP8266_ELRS_TX and then said MODULE_ESP8266_INTERNAL_TX is good. I'm confused enough on my own!

ohh ... sorry ... copied the worng ... MODULE_ESP8266_ELRS_TX  is good :)

Reply:
> Try to think of it as esp32 getting the last word. :)

haha hard to beat that argument :D so go ahead :)

Reply:
> ESP8266WiFi.h It also just more common to include standard system headers before jumping into defining variables, etc.

well, yes ... but not with Arduino, here we often can keep things to have it cleaner ...
so, again, is this needed or not?

Reply:
ESP8266WiFi.h needs to be included before we can use IPAddress.  I guess if you want, I can split up the includes and put most of them back where they were, but it still makes more sense to me this way.

Reply:
look at the old code ... it was (seemingly) the same situation !!
so, again, did you had to do it becasue you got a compile error, or did you do it presumptiously

Reply:
Compile error for esp8266.  That's why I said "needs".  I moved the rest of the includes as well to keep them all together.  Do you want them split up?

Reply:
The comment on the include also explained all of this:
#include <ESP8266WiFi.h> // need this before WiFi settings section

Reply:
that's strange it is needed for the esp8266 and not the esp32 ... I guess I will need to analyse retrospectively what's really going on here
so leave it as it is, I will go over it again after merge